align_system/algorithms/decision_flow_adm/math_reason_stage_component.py

```diff
@@ -0,0 +1,179 @@
+                    attr_value = kdma_value_dict.get(attribute.kdma)
+                    if attr_value is not None and attr_value >= 0.5:
+                        bias_parts.append(f"high {attribute.name}")
+                    elif attr_value is not None:
+                        bias_parts.append(f"low {attribute.name}")
+                    else:
+                        bias_parts.append(f"low {attribute.name}")  # Conservative default
+                target_bias = f"Focus on {', '.join(bias_parts)} when making decisions."
+            else:
+                target_bias = "Make decisions based on medical triage best practices."
+
+            dialog = []
+            if self.system_prompt_template is not None:
+                system_prompt = call_with_coerced_args(
+                    self.system_prompt_template,
+                    {
+                        'alignment_target': alignment_target,
+                        'target_attributes': target_attributes
+                    }
+                )
+                dialog.insert(0, DialogElement(role='system',
+                                              content=system_prompt,
+                                              tags=['decisionflow_system_prompt']))
+
+            prompt = call_with_coerced_args(
+                self.prompt_template,
+                {
+                    'objective': structure["Objective Function"],
+                    'attribute': attribute,
+                    'variable': structure["Decision Variables"],
+                    'constraints': structure["Constraints"],
+                    'choice': formatted_choices,
+                    'target_bias': target_bias
+                },
+            )
+
+            dialog.append(DialogElement(role='user',
+                                       content=prompt,
+                                       tags=['decisionflow_math_reason']))
+
+            output_schema = call_with_coerced_args(
+                self.output_schema_template,
+                {})
+
+            dialog_prompt = self.structured_inference_engine.dialog_to_prompt(dialog)
+
+            response = self.structured_inference_engine.run_inference(
+                dialog_prompt,
+                output_schema
+            )
+
+            # Parse response to get chosen action
+            reasoning = response.get('Reasoning', '')
+            answer_idx = response.get('Answer', 0)
+
+            # Validate answer index
+            if not isinstance(answer_idx, int) or answer_idx < 0 or answer_idx >= len(actions):
+                log.warning(f"Invalid answer index {answer_idx}, defaulting to 0")
+                answer_idx = 0
+
+            chosen_action = actions[answer_idx]
+
+            # Add reasoning as justification if possible
+            if hasattr(chosen_action, 'justification') and chosen_action.justification is None:
+                chosen_action.justification = reasoning
```

Comment:
This logic looks like it's from our pre-outlines ADM code.  There's probably a way to set up the schema such that we just have it pick the answer by it's text rather than the answer index.  But that's probably different from how the original decision flow is set up, so let's consider that for a follow-up MR.